src/main/java/fi/aalto/cs/apluscourses/intellij/utils/CourseFileManager.java

```diff
@@ -192,9 +203,18 @@ private void loadFromJsonObject(@NotNull JSONObject jsonObject) throws IOExcepti
     Iterable<String> moduleNames = modulesObject::keys;
     for (String moduleName : moduleNames) {
       JSONObject moduleObject = modulesObject.getJSONObject(moduleName);
+
       String moduleId = moduleObject.getString(MODULE_ID_KEY);
-      ZonedDateTime downloadedAt
-          = ZonedDateTime.parse(moduleObject.getString(MODULE_DOWNLOADED_AT_KEY));
+
+      String downloadedAtStr = moduleObject.optString(MODULE_DOWNLOADED_AT_KEY);
+      ZonedDateTime downloadedAt;
+      if (downloadedAtStr == null || downloadedAtStr.trim().isEmpty()) {
+        logger.error("Module " + moduleName + " missing 'downloadedAt' in course file");
+        downloadedAt = Instant.EPOCH.atZone(ZoneId.systemDefault());
```

Comment:
`ZonedDateTime.now()`?

Reply:
This is same misunderstanding that I had: https://github.com/Aalto-LeTech/intellij-plugin/pull/390#discussion_r488571846

`downloadedAt` is NOT set to the current date (and it shouldn't). Instant.EPOCH refers to `1970-01-01`, that is "the instant of epoch", not "this instant second in epoch time" as both of us, apparently, read it. :D